src/InstrumentationEngine.Lib/refcount.h

```diff
@@ -147,10 +147,42 @@ namespace MicrosoftInstrumentationEngine
         static class CRefRBMap g_map;
         static CRITICAL_SECTION g_crst;
 
-        static void RecordCreation(CRefCount *pCRefCount);
-        static void RecordDestruction(CRefCount *pCRefCount);
-        static void RecordAddRef(CRefCount *pCRefCount);
-        static void RecordRelease(CRefCount *pCRefCount);
+        static void __forceinline RecordCreation(CRefCount* pCRefCount)
+        {
+            if (g_fRecordingInitialized)
```

Comment:
I'm assuming this is still wrapped in an ifdef DEBUG? Also does the compiler honor __forceinline in Debug builds?

Reply:
Yes, in the caller.

It seems a little faster __forceinline, but the callstack doesn't indicate that it is inlined. It is hard to tell. I'll also try creating a macro.

Reply:
OK, so I wrapped the functions in a macro instead, and it seems to save about 1 second of time for the InstrEngineTests overall when recording is turned off.

After more testing, it does turn out that having recording on causes a relatively small overhead. The InstrEngine tests run about 1/2 second slower in total when recording is turned on. That being said, these tests are pretty small and create/destroy a relatively small number of objects. When we talked offline, I thought that the tests had a much larger overhead because I had accidentally had a debugger attached to one of them... oops.

Anyway, I'm not going to try to squeeze any more out of this, it is probably as good as I can do.